packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-in-app-1.test.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  addNotificationChannel,
+  amplifyPull,
+  amplifyPushAuth,
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+  getAppId,
+  getProjectMeta,
+  initJSProjectWithProfile,
+  removeAnalytics,
+  addPinpointAnalytics,
+  removeNotificationChannel,
+  getBackendAmplifyMeta,
+  amplifyStatus,
+  getTeamProviderInfo,
+  removeAllNotificationChannel,
+} from '@aws-amplify/amplify-e2e-core';
+import {
+  expectLocalAndPulledAwsExportsMatching,
+  expectLocalAndPulledBackendAmplifyMetaMatching,
+  expectLocalAndPulledBackendConfigMatching,
+  getShortId,
+} from '../import-helpers';
+
+describe('notification category compatibility test', () => {
+  const testChannelSelection = 'In-App Messaging';
+  const envName = 'test';
+  const projectPrefix = `notificationCompatibility`.substring(0, 19);
+  const projectSettings = {
+    name: projectPrefix,
+    disableAmplifyAppCreation: false,
+    envName,
+  };
+
+  let projectRoot: string;
+  let pullTestProjectRoot: string;
+
+  beforeEach(async () => {
+    projectRoot = await createNewProjectDir(projectPrefix);
+  });
+
+  afterEach(async () => {
+    await removeAnalytics(projectRoot, {});
+    await deleteProject(projectRoot);
+    deleteProjectDir(projectRoot);
+    if (pullTestProjectRoot) {
+      deleteProjectDir(pullTestProjectRoot);
+    }
+  });
+
+  it(`works with existing pinpoint that hasn't pushed`, async () => {
+    const pinpointResourceName = `${projectPrefix}${getShortId()}`;
+    await initJSProjectWithProfile(projectRoot, projectSettings);
+
+    const appId = getAppId(projectRoot);
+    expect(appId).toBeDefined();
+
+    // BEGIN - SETUP PINPOINT BUT DON'T PUSH (see analytics.test.ts)
+    await addPinpointAnalytics(projectRoot, pinpointResourceName);
+
+    // SETUP NOTIFICATIONS CHANNEL BUT DON'T PUSH (IN-APP MESSAGING)
```

Comment:
super nit: not a fan of caps comments

Reply:
+1. additionally not a fan of comments. can the code be structured in such a way (or is it already) that one can read control flow from function names ?